Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -393,37 +336,92 @@ class Navigator {
   /// @param [in] stepper Stepper in use
   template <typename propagator_state_t, typename stepper_t>
   void preStep(propagator_state_t& state, const stepper_t& stepper) const {
+    ACTS_VERBOSE(volInfo(state) << "Entering navigator::preStep.");
+
     // Check if the navigator is inactive
-    if (inactive(state, stepper)) {
+    if (inactive(state)) {
       return;
     }
 
-    // Call the navigation helper prior to actual navigation
-    ACTS_VERBOSE(volInfo(state) << "Entering navigator::preStep.");
+    bool reinitialized = false;
+    // Check next navigation candidate
+    while (state.navigation.candidateIndex !=
+           state.navigation.candidates.size()) {
+      // Screen output how much is left to try
+      ACTS_VERBOSE(volInfo(state)
+                   << (state.navigation.candidates.size() -
+                       state.navigation.candidateIndex)
+                   << " out of " << state.navigation.candidates.size()
+                   << " surfaces remain to try.");
+
+      const auto& candidate = state.navigation.candidate();
+      const auto& intersection = candidate.intersection;
+      const Surface& surface = *intersection.representation();
+      BoundaryCheck boundaryCheck = candidate.boundaryCheck;
+
+      // Screen output which surface you are on
+      ACTS_VERBOSE(volInfo(state) << "Next surface candidate will be "
+                                  << surface.geometryId());
+
+      // Estimate the surface status
+      auto surfaceStatus = stepper.updateSurfaceStatus(
+          state.stepping, surface, intersection.index(),
+          state.options.direction, boundaryCheck,
+          state.options.surfaceTolerance, logger());
+
+      if (surfaceStatus == IntersectionStatus::onSurface) {
+        ACTS_ERROR(volInfo(state) << "We are on surface before trying to reach "
+                                     "it. This should not happen. Good luck.");
```

Comment:
this is one of the errors I log right now @paulgessinger. I guess we have to either make sure it never fires or to reduce the level?